### DISCUSSION TITLE
Disable edges behind nodes

### DISCUSSION THREAD

**Discussion by xxxLukskyxxx** [xyflow/xyflow#1141] [Q&A]
Hello everybody,

it is possible to disable edges behind nodes? 
If there is no possibility to disable this, it may be an helpful feature for the future, because a bigger chart is very difficult to understand if the edges are covered by nodes. 

Thanks for your help! :slightly_smiling_face:

**Comment by harshagarwalnyu** [comment]
React Flow renders edges in a separate SVG layer that sits **below** the nodes layer by default. There are two ways to control this:

**Option 1 — `elevateEdgesOnSelect` (built-in prop)**
```jsx
<ReactFlow
  nodes={nodes}
  edges={edges}
  elevateEdgesOnSelect  // selected edges render above all nodes
/>
```
This temporarily elevates selected edges. Good for interactive inspection but doesn't fix permanently-hidden edges.

**Option 2 — Per-edge `zIndex` prop**
```tsx
const edges = [
  {
    id: 'e1-2',
    source: '1',
    target: '2',
    zIndex: 1,  // renders above nodes (nodes default to z=0)
  },
];
```
Setting `zIndex` on individual edges lifts them above the node layer. Set it on all edges if you always want edges visible:

```tsx
const elevatedEdges = edges.map(e => ({ ...e, zIndex: 1 }));
```

**Option 3 — `elevateNodesOnSelect` + always-elevated edges**
If your issue is specifically about selected nodes covering adjacent edges:
```jsx
<ReactFlow
  elevateNodesOnSelect  // selected node renders above others
  edges={edges.map(e => ({ ...e, zIndex: 1 }))}
/>
```

**Recommendation:** For large, dense graphs where edge visibility is a core concern, use `zIndex: 1` on all edges as a default. Combine with `elevateNodesOnSelect` so selected nodes still appear above their (now-elevated) edges.

**Comment by tisoap** [comment]
I'm the creator of [react-flow-smart-edge](https://github.com/tisoap/react-flow-smart-edge) , a library that uses using A* pathfinding to route edges around other nodes, avoiding intersections. It has smart equivalents for every type of edge React Flow has. It's been updated to work with React Flow 12.
